### PR TITLE
Re-enable ~20 xunit analyzers

### PR DESCRIPTION
**Pull request by stephentoub**
All but one of these had no associated warnings firing.  I fixed one warning to be able to re-enable another analyzer.

cc: @ViktorHofer

### CODE REVIEW

File: src/System.Memory/tests/Span/GcReporting.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ public static class SpanGcReportingTests
         /// This is a simple sanity test to check that GC reporting for Span is not completely broken, it is not meant to be
         /// comprehensive.
         /// </summary>
+        [Theory]
```
Comment (ViktorHofer) [inline src/System.Memory/tests/Span/GcReporting.cs]: any idea why this wasn't caught with the existing enabled analyzers? I thought I fixed all of these.

Comment (stephentoub) [reply src/System.Memory/tests/Span/GcReporting.cs]: Don't know.  This was triggering https://xunit.net/xunit.analyzers/rules/xUnit1008, which was disabled.

Comment (ViktorHofer) [reply src/System.Memory/tests/Span/GcReporting.cs]: Super weird. Means that the analyzers are not guaranteed to be correct.

**Comment by ViktorHofer** [comment]
These are the remaining ones. Some might be buggy, some might not bring any value...

```
xUnit1006	Theory methods should have parameters
xUnit1010	The value is not convertible to the method parameter type
xUnit1016	MemberData must reference a public member
xUnit1019	MemberData must reference a member providing a valid data type
xUnit1024	Test methods cannot have overloads
xUnit1025	InlineData should be unique within the Theory it belongs to
xUnit1026	Theory methods should use all of their parameters
xUnit2000	Constants and literals should be the expected argument
xUnit2001	Do not use invalid equality check
xUnit2002	Do not use null check on value type
xUnit2003	Do not use equality check to test for null value
xUnit2004	Do not use equality check to test for boolean conditions
xUnit2005	Do not use identity check on value type
xUnit2006	Do not use invalid string equality check
xUnit2007	Do not use typeof expression to check the type
xUnit2008	Do not use boolean check to match on regular expressions
xUnit2009	Do not use boolean check to check for substrings
xUnit2012	Do not use Enumerable.Any() to check if a value exists in a collection
xUnit2013	Do not use equality check to check for collection size.
xUnit2015	Do not use typeof expression to check the exception type
xUnit2017	Do not use Contains() to check if a value exists in a collection
xUnit2018	Do not compare an object's exact type to an abstract class or interface
```

File: src/System.Memory/tests/Span/GcReporting.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Memory/tests/Span/GcReporting.cs]: Looking at the xunit analyzer code for 1013, it only fires if the public method in question is on a class with other tests (a reasonable heuristic)... this class doesn't have any others.

**Pull request merged**